src/server/shared/Logging/Log.h

```diff
@@ -121,51 +140,31 @@ inline bool Log::ShouldLog(std::string const& type, LogLevel level) const
     return logLevel != LOG_LEVEL_DISABLED && logLevel <= level;
 }
 
-inline void Log::outMessage(std::string const& filter, LogLevel level, const char * str, ...)
-{
-    va_list ap;
-    va_start(ap, str);
-
-    vlog(filter, level, str, ap);
-
-    va_end(ap);
-}
-
 #define sLog Log::instance()
 
-#if PLATFORM != PLATFORM_WINDOWS
-#define TC_LOG_MESSAGE_BODY(filterType__, level__, ...)                 \
-        do {                                                            \
-            if (sLog->ShouldLog(filterType__, level__))                 \
-                sLog->outMessage(filterType__, level__, __VA_ARGS__);   \
-        } while (0)
-#else
-#define TC_LOG_MESSAGE_BODY(filterType__, level__, ...)                 \
-        __pragma(warning(push))                                         \
-        __pragma(warning(disable:4127))                                 \
-        do {                                                            \
-            if (sLog->ShouldLog(filterType__, level__))                 \
-                sLog->outMessage(filterType__, level__, __VA_ARGS__);   \
-        } while (0)                                                     \
-        __pragma(warning(pop))
-#endif
+template<typename... Args>
+inline void RequestLog(std::string const& filter, LogLevel const level, const char* fmt, Args const&... args)
```

Comment:
are arguments evaluated even if sLog->ShouldLog() returns false ? if so, then this is not an acceptable change due to performance reasons